client/components/site-selector/index.jsx

```diff
@@ -293,11 +301,20 @@ class SiteSelector extends Component {
 		return siteElements;
 	}
 
+	setHighlightedSiteRef = isHighlighted => component => {
+		if ( isHighlighted && component ) {
```

Comment:
Have we tried calling this.isHighlighted() a second time so we can avoid generating functions in render?

Reply:
Yeah, I spent some time thinking about how to avoid generating functions in render. However, as you can see, `this.isHighlighted` accepts `siteId`. So, whether a site component is highlighted or not depends on which siteId it is. We would have to create the same number of functions as there are sites in order to not have to supply `siteId`.

Or am I missing something?

Reply:
Ah ok there are two usages, so we'd need one function for all sites, and another for site. I believe site should be available as a prop? Still might be worth giving it a try.

Reply:
Well, site is not available as a prop because this component (`SiteSelector`) renders *all* the sites. A site is supplied to the `renderSite` function from `renderSites`.

Reply:
ah got it, the rest of the code looked fine btw, so if manual tests are good, feel free to 🚢